src/website/app/pages/Theming/theming.md

```diff
@@ -1,63 +1,145 @@
+
+function App() {
+  return (
+    <ThemeProvider>
+      <Button>
+        Hello World
+      </Button>
+    </ThemeProvider>
+  );
+}
+
+render(<App />, document.getElementById('app'));
 ```
 
-The [ThemeProvider](/components/theme-provider)(s) in
-your app provides the theme to other Mineral UI and/or Glamorous
-components within that ThemeProvider. Your app
-[must have a ThemeProvider at its root](/getting-started) and can
-optionally nest additional ThemeProviders to apply a custom theme to sections
-of your app. Nested ThemeProviders shallowly merge their theme
-with the parent theme.
+### Theme a portion of your app
+
+ThemeProviders may be nested in order to apply a custom theme to a portion of your app.  Nested ThemeProviders shallowly merge their theme with the parent theme.  The theme itself is a simple shallow object of variables that are shared across components.  [See the default mineralTheme below for an example](#theming-theme-structure).
+
+```jsx
+<ThemeProvider>
+  <ThemeProvider theme={{ color_primary: 'darkgray' }}>
+    <nav>Navigation<nav>
+  </ThemeProvider>
+  <main>The main part of your app</main>
+</ThemeProvider>
+```
 
-The theme itself (see the default mineralTheme below for an example) is
-a simple shallow object of variables that are shared across components.
+### Theme a component
 
-Each component can also have a “theme”, which is not a file, but rather
-a set of variables available to override default values. E.g., if
-Mineral UI’s [Button](/components/button) component looked like this:
+Each component has a set of component-level theme variables that may be overridden to adjust styles on a per component basis.  These are documented on the individual component pages.  e.g. [Button theme variables](/components/button/#theme-variables)
+
+To theme a component, use [createThemedComponent](#theming-api) as shown below.  It is effectively the same as wrapping your component with a ThemeProvider.
 
 ```jsx
-const Button = createStyledComponent('button', props => ({
-  color: props.theme.Button_color || props.theme.color_primary
-}));
+import { createThemedComponent } from 'mineral-ui/themes';
+
+const MyButton = createThemedComponent(Button, {
+  Button_backgroundColor: 'tomato'
+});
 ```
 
-The themes distributed as part of Mineral UI include a value for
-`color_primary` but do not include a value for `Button_color`. In
-our component code, we leave the component-level variable
-`Button_color` as a hook for you to define if you’d like.
-Component-level theme variables start with the capitalized component
-name to differentiate from the global variables. When you’d like to
-override the Mineral UI theme at a component level in your app, you can
-use [createThemedComponent](/components/utils#create-themed-component):
+### Create your own theme
+
+Use [createTheme](#theming-api) in order to create a custom theme.  Once created, this theme can be applied using a ThemeProvider.
```

Comment:
Maybe we should point them to https://415-themes-deux--mineral-ui.netlify.com/color/#color-ramps to illustrate which colors can be a baseColor?

Reply:
The `createTheme` API docs on this page does just that.